lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -125,33 +125,36 @@ function dosomething_northstar_update_user($user, $password = null) {
 }
 
 /**
- * Get user profile data from Northstar for the given Drupal ID.
+ * Get user profile data from Northstar for the given ID.
  *
- * @param int $drupal_id Drupal user id.
+ * @param int $id Drupal user id.
+ * @param string $type The index type of the ID
  * @return object
  */
-function dosomething_northstar_get_northstar_user($drupal_id) {
-  $northstar_user_info = cache_get('northstar_user_info_' . $drupal_id, 'cache_dosomething_northstar');
+function dosomething_northstar_get_northstar_user($id, $type = 'drupal_id') {
+  $northstar_user_info = NULL;
+
+  if ($type === 'drupal_id') {
+    $northstar_user_info = cache_get('northstar_user_info_' . $id, 'cache_dosomething_northstar');
+  }
 
   if ($northstar_user_info) {
     $northstar_user_info = $northstar_user_info->data;
   }
   else {
     $client = _dosomething_northstar_build_http_client();
-
-    $northstar_user_info = drupal_http_request($client['base_url'] . '/users/drupal_id/' . $drupal_id, array(
+    $response = drupal_http_request($client['base_url'] . '/users/' . $type . '/' . $id, [
       'headers' => $client['headers'],
       'method' => 'GET',
-      ));
-
-      $northstar_user_info = $northstar_user_info->data;
-
-      if (!empty($northstar_user_info->error)) {
-        $error = sprintf("Error fetching Northstar user data, uid=%d: '%s'", $drupal_id, $northstar_user_info->error);
-        watchdog_exception('northstar', new Exception($error));
-      } else {
-        cache_set('northstar_user_info_' . $drupal_id, $northstar_user_info, 'cache_dosomething_northstar', REQUEST_TIME + 60*60*24);
-      }
+    ]);
+    $northstar_user_info = $response->data;
+
+    if (!empty($northstar_user_info->error)) {
+      $error = sprintf("Error fetching Northstar user data, uid=%d: '%s'", $id, $northstar_user_info->error);
+      watchdog_exception('northstar', new Exception($error));
+    } else {
+      cache_set('northstar_user_info_' . json_decode($northstar_user_info)->data->drupal_id, $northstar_user_info, 'cache_dosomething_northstar', REQUEST_TIME + 60*60*24);
```

Comment:
we have a lot of ID's (4 now?) and if we do more social stuff it'll grow, I think we're better off finding a more dynamic solution later on